iceoryx2/src/port/publisher.rs

```diff
@@ -1018,6 +1022,31 @@ impl<Service: service::Service, Payload: Debug, UserHeader: Debug>
         )
     }
 }
+
+impl<Service: service::Service, UserHeader: Debug>
+    Publisher<Service, [CustomPayloadMarker], UserHeader>
+{
+    /// # Safety
+    ///
+    ///  * slice_len != 1 only when payload TypeVariant == Dynamic
+    ///  * The number_of_elements in the [`Header`](crate::service::header::publish_subscribe::Header)
+    ///     is set to `slice_len`
+    ///  * The [`SampleMutUninit`] will contain `slice_len` * `MessageTypeDetails::payload.size`
+    ///     elements of type [`CustomPayloadMarker`].
+    #[doc(hidden)]
+    pub unsafe fn loan_custom_payload(
+        &self,
+        slice_len: usize,
+    ) -> Result<
+        SampleMutUninit<Service, [MaybeUninit<CustomPayloadMarker>], UserHeader>,
+        PublisherLoanError,
+    > {
+        // TypeVariant::Dynamic == slice and only here it makes sense to loan more than one element
+        debug_assert!(slice_len == 1 || self.payload_type_variant() == TypeVariant::Dynamic);
+
+        self.loan_slice_uninit_impl(slice_len, self.payload_size * slice_len)
```

Comment:
I have the feeling there is still not a clear definition what `payload` means and it is used in different contexts with a different meaning, resulting in confusion at some point.

No need to change it now but we need to have unambiguous terminology, else this will haunt us in regular intervals.

Reply:
Just out of my head:
* payload == all bytes that belong to the user produced data
* slice can be a form of payload
* a slice consists of 1..N elements, therefore we have in this context an element_size?